src/boot/efi/boot.c

```diff
@@ -1185,6 +1189,26 @@ static BootEntry* boot_entry_free(BootEntry *entry) {
 
 DEFINE_TRIVIAL_CLEANUP_FUNC(BootEntry *, boot_entry_free);
 
+static bool parse_console_color(const char *v, uint8_t *color_out){
+        const char *color_names[] = {
+                "black", "blue", "green", "cyan",
+                "red", "magenta", "brown", "lightgray",
+                "bright", "darkgray", "lightblue", "lightgreen",
+                "lightcyan", "lightred", "lightmagenta", "yellow", "white"};;
+        uint8_t color_values[] = {
+                EFI_BLACK, EFI_BLUE, EFI_GREEN, EFI_CYAN,
+                EFI_RED, EFI_MAGENTA, EFI_BROWN, EFI_LIGHTGRAY,
+                EFI_BRIGHT, EFI_DARKGRAY, EFI_LIGHTBLUE, EFI_LIGHTGREEN,
+                EFI_LIGHTCYAN, EFI_LIGHTRED, EFI_LIGHTMAGENTA, EFI_YELLOW, EFI_WHITE};
+        for(int i = 0; i < 17; i++) {
+                if(streq8(v, color_names[i])) {
+                        *color_out = color_values[i];
+                        return true;
+                }
+        }
+        return false;
+}
```

Comment:
so what you are implementing here the stuff we usually to with DEFINE_STRING_TABLE_LOOKUP(), at least in our userspace codebase. 

I think we should not reimplement this here, but just do it properly, i.e. move the relevant bits (which might be all of them) from src/basic/string-table.h to src/fundamental/string-table.h. Sooner or later we have to do that anyway I guess. (Moving this would effectively mean it's accessible from both efi mode and userspace. That's what src/fundamental/*.h is for: code that is shared by both worlds).